tensorboard/webapp/webapp_data_source/tb_feature_flag_data_source.ts

```diff
@@ -40,62 +34,39 @@ export class QueryParamsFeatureFlagDataSource
   constructor(readonly queryParams: QueryParams) {}
 
   getFeatures(enableMediaQuery: boolean = false) {
-    const params = this.queryParams.getParams();
     // Set feature flag value for query parameters that are explicitly
     // specified. Feature flags for unspecified query parameters remain unset so
     // their values in the underlying state are not inadvertently changed.
-    const featureFlags: Partial<FeatureFlags> = enableMediaQuery
-      ? this.getPartialFeaturesFromMediaQuery()
-      : {};
-    if (params.has(EXPERIMENTAL_PLUGIN_QUERY_PARAM_KEY)) {
-      featureFlags.enabledExperimentalPlugins = params.getAll(
-        EXPERIMENTAL_PLUGIN_QUERY_PARAM_KEY
-      );
-    }
-    if (params.has('tensorboardColab')) {
-      featureFlags.inColab = params.get('tensorboardColab') === 'true';
-    }
-    if (params.has(SCALARS_BATCH_SIZE_PARAM_KEY)) {
-      featureFlags.scalarsBatchSize = Number(
-        params.get(SCALARS_BATCH_SIZE_PARAM_KEY)
-      );
-    }
-
-    if (params.has(ENABLE_COLOR_GROUP_QUERY_PARAM_KEY)) {
-      featureFlags.enabledColorGroup =
-        params.get(ENABLE_COLOR_GROUP_QUERY_PARAM_KEY) !== 'false';
-    }
-
-    if (params.has(ENABLE_COLOR_GROUP_BY_REGEX_QUERY_PARAM_KEY)) {
-      featureFlags.enabledColorGroupByRegex =
-        params.get(ENABLE_COLOR_GROUP_BY_REGEX_QUERY_PARAM_KEY) !== 'false';
-    }
-
-    if (params.has(ENABLE_DARK_MODE_QUERY_PARAM_KEY)) {
-      featureFlags.defaultEnableDarkMode =
-        params.get(ENABLE_DARK_MODE_QUERY_PARAM_KEY) !== 'false';
-    }
-
-    if (params.has(ENABLE_LINKED_TIME_PARAM_KEY)) {
-      featureFlags.enabledLinkedTime =
-        params.get(ENABLE_LINKED_TIME_PARAM_KEY) !== 'false';
-    }
-
-    if (params.has(ENABLE_CARD_WIDTH_SETTING_PARAM_KEY)) {
-      featureFlags.enabledCardWidthSetting =
-        params.get(ENABLE_CARD_WIDTH_SETTING_PARAM_KEY) !== 'false';
-    }
+    const featureFlags: Partial<Record<keyof FeatureFlags, FeatureFlagType>> =
+      enableMediaQuery ? this.getPartialFeaturesFromMediaQuery() : {};
+    Object.entries(FeatureFlagMetadataMap).forEach(
```

Comment:
Can this also be refactored into something reusable? I think you were already intending on reusing getFeatureValue() but maybe we don't have to stop there. Especially given how this mangles all type information, would be nice to just have to write that code once.